File: 2274-keep-multiplying-found-values-by-two/keep-multiplying-found-values-by-two.java
class Solution {
    public int findFinalValue(int[] nums, int k) {
              int bits = 0;
        for (int n : nums) {
            if (n % k != 0) continue;
            n /= k;
            if ((n & (n - 1)) == 0)
                bits |= n;
        }
        bits++;
        return k * (bits & -bits);
     
    }
}

File: 2274-keep-multiplying-found-values-by-two/keep-multiplying-found-values-by-two.cpp
class Solution {
public:
    int findFinalValue(vector<int>& nums, int org) {
       int i=1;
       while(i)
       {
         if(find(nums.begin(),nums.end(),org)!=nums.end())
         org*=2;
         else
         i=0;
  
       }
       return org;
    }
};